doc/cutting-a-release.md

```diff
@@ -0,0 +1,135 @@
+# Creating a new release of google-cloud-cpp
+
+Unless there are no changes, we create releases for `google-cloud-cpp` every
+4 weeks, or if there is a major announcement or change to the status of one
+of the libraries (like reaching the "Alpha" or "Beta" milestone).
+
+The intended audience of this document are developers in the `google-cloud-cpp`
+project that need to create a new release. The audience is expected to be
+familiar with the project itself, [git][git-docs], [GitHub][github-guides],
+[semantic versioning](https://semver.org).
+
+## Preparing for a release
+
+First you should collect and update the release notes for the project. Prepare
+a pull request (PR) with the necessary changes to the README files in each
+project.
+
+Assuming you are working on your own fork of the `google-cloud-cpp` project,
+and `upstream` points to the `GoogleCloudPlatform/google-cloud-cpp` remote,
+these commands should be useful in identifying important changes:
+
+```bash
+# Summarize the output of this into google/cloud/bigtable/README.md
+git log upstream/master -- google/cloud/bigtable
+```
+
+```bash
+# Summarize the output of this into google/cloud/storage/README.md
+git log upstream/master -- google/cloud/bigtable
+```
+
+Do not forget to update `google/cloud/README.md` too.
+
+It is not recommended that you create the release branch before this PR is
+ready, but in some circumstances it might be needed, for example, if a large
+change that could destabilize the release is about to be merged, or if we want
+to create the release at an specific point in the revision history.
+
+## Create the release branch
+
+Through this document we will use this variable to represent the release name:
+
+```bash
+export REL="v0.N"
+```
+
+If you decide to cut&paste the commands below, make sure that variable has the
+actual release value, e.g. `v0.5` or `v0.7`, and not the generic `N`.
+
+Clone the main repository to create the branch:
+
+```sh
+git clone git@github.com:GoogleCloudPlatform/google-cloud-cpp.git releases
+cd releases
+git checkout -b "${REL}.x"
+```
+
+Modify the CMake script to indicate this is a release:
+
+```bash
+sed -i 's/set(GOOGLE_CLOUD_CPP_IS_RELEASE "")/set(GOOGLE_CLOUD_CPP_IS_RELEASE "yes")/' google/cloud/CMakeLists.txt 
```

Comment:
Can you pull this value from an environment variable instead?

Reply:
We don't want that.  Regardless of how the customers build a released branch, they should get the release version tags.